internal/dag/builder.go

```diff
@@ -126,17 +126,18 @@ func (b *builder) lookupTCPService(m meta, port intstr.IntOrString, strategy str
 		return nil
 	}
 }
-func (b *builder) lookupService(m meta, port intstr.IntOrString, strategy string, hc *ingressroutev1.HealthCheck) Service {
+func (b *builder) lookupService(m meta, port intstr.IntOrString, strategy string, hc *ingressroutev1.HealthCheck, uv *ingressroutev1.UpstreamValidation) Service {
 	if port.Type != intstr.Int {
 		// can't handle, give up
 		return nil
 	}
 	sm := servicemeta{
-		name:        m.name,
-		namespace:   m.namespace,
-		port:        int32(port.IntValue()),
-		strategy:    strategy,
-		healthcheck: healthcheckToString(hc),
+		name:               m.name,
+		namespace:          m.namespace,
+		port:               int32(port.IntValue()),
+		strategy:           strategy,
+		healthcheck:        healthcheckToString(hc),
+		upstreamvalidation: upstreamValidationToString(uv),
```

Comment:
Sorry, this is not correct. The upstream validation is not a property of the service, it is a propery of the cluster -- the path from the route to the service.